R/read_sourmash_outputs.R

```diff
@@ -0,0 +1,156 @@
+#' read_compare_csv
+#'
+#' @param file path to file output by sourmash compare using the with the --csv flag
+#' @param sample_to_rownames Boolean indicating whether sample names should be added to the tibble as a column or a rowname
+#' @param ... Arguments passed to readr::read_csv()
+#'
+#' @return A tibble.
+#' @export
+#'
+#' @examples
+#' read_compare_csv("")
+read_compare_csv <- function(file, sample_to_rownames = F, ...){
+  # guard the use of required packages
+  if (!requireNamespace("readr", quietly = TRUE)) {
+    stop(
+      "Package \"readr\" must be installed to use this function.",
+      call. = FALSE
+    )
+  }
+
+  if (!requireNamespace("dplyr", quietly = TRUE)) {
+    stop(
+      "Package \"dplyr\" must be installed to use this function.",
+      call. = FALSE
+    )
+  }
+
+  if(sample_to_rownames == F){
+  compare_df <- readr::read_csv(file, ...) %>%
+    dplyr::mutate(sample = colnames(.), .before = everything())
+  } else if(sample_to_rownames == T){
+
+    if (!requireNamespace("tibble", quietly = TRUE)) {
+      stop(
+        "Package \"tibble\" must be installed to make sample names into row names. Please either install \"tibble\" or set sample_to_rownames to FALSE.",
+        call. = FALSE
+      )
+    }
+
+    compare_df <- readr::read_csv(file, ...) %>%
+      dplyr::mutate(sample = colnames(.), .before = everything()) %>%
+      tibble::column_to_rownames("sample")
+  }
+  return(compare_df)
+}
+
+#' read gather
+#'
+#' @param file Path to CSV file or files output by sourmash gather.
+#' @param intersect_bp_threshold Integer. Gather matches must have an intersect_bp greater than or equal to this value.
+#' @param ... Arguments passed to read_csv().
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' read_gather()
+read_gather <- function(file, intersect_bp_threshold, ...){
+  if (!requireNamespace("readr", quietly = TRUE)) {
+    stop(
+      "Package \"readr\" must be installed to use this function.",
+      call. = FALSE
+    )
+  }
+
+  if (!requireNamespace("dplyr", quietly = TRUE)) {
+    stop(
+      "Package \"dplyr\" must be installed to use this function.",
+      call. = FALSE
+    )
+  }
+
+  if(length(files) > 1){
+
+    if (!requireNamespace("purrr", quietly = TRUE)) {
+      stop(
+        "Package \"purrr\" must be installed to read multiple files at a time. Please either install \"purrr\" or read one file at a time.",
+        call. = FALSE
+      )
+    }
+
+    # allow the function to read multiple files at once
+    taxonomy_annotate_df <- file %>%
+      purrr::map_dfr(readr::read_csv, col_types = "ddddddddcccddddcccddcddlddddl", ...) %>%
```

Comment:
What is this `col_types` referring to? A string?

Reply:
ya i hate this syntax. Each character in the string corresponds to one column (`c`= character, `d` = digits, `l` = logical). I use this because it there is an empty file, or one that happens to have a digits in a character column, `bind_rows`/purrr concatenation is blocked because they types don't match.